base/arraymath.jl

```diff
@@ -67,15 +60,21 @@ end
 
 for f in (:.+, :.-, :.*, :./, :.\, :.^, :.÷, :.%, :.<<, :.>>, :div, :mod, :rem, :&, :|, :$)
     @eval begin
-        function ($f){T}(A::Number, B::AbstractArray{T})
-            F = similar(B, promote_array_type($f,typeof(A),typeof(B)))
+        function ($f)(A::Number, B::AbstractArray)
+            P = promote_op($f, typeof(A), eltype(B))
+            T = promote_array_type($f, typeof(A), eltype(B), P)
+            T === Any && return [($f)(A, b) for b in B]
+            F = similar(B, T)
```

Comment:
Yes, so once the dot-operators are turned into `broadcast`s we will _consistently_ have `Array` return types (unless we adapt `broadcast` as is already done to some extent for `SparseMatrixCSC`, which we probably should do). Having `promote_op` not only decide the element type, but also the container type _might_ be an issue, but only temporarily anyway.